src/util/circular_buffer.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2018-2022 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#include "RG.h"
+#include "rmalloc.h"
+#include "circular_buffer.h"
+
+// circular buffer structure
+// the buffer is of fixed size
+// items are removed by order of insertion, similar to a queue
+struct _CircularBuffer {
```

Comment:
what circular about this buffer? it is look like a queue

Reply:
line 60
```c
    // advance write position
    // circle back if write reached the end of the buffer
    cb->write += cb->item_size;
    if(cb->write >= cb->data + cb->item_size * cb->item_cap) {
        cb->write = cb->data;
    }
```

Reply:
TODO: add unit test for circular buffer